packages/angular-cli/tasks/test.ts

```diff
@@ -13,7 +13,8 @@ export default Task.extend({
     const projectRoot = this.project.root;
     return new Promise((resolve) => {
       const karma = requireDependency(projectRoot, 'karma');
-      const karmaConfig = path.join(projectRoot, this.project.ngConfig.config.test.karma.config);
+      const karmaConfigFile = this.project.ngConfig.config.test.karma[options.config];
```

Comment:
I don't think this should be looking the path up in the config file. The default file location will suffice, just need to update the content of that file if different configuration is needed. If an alternative (read: secondary) config file is needed, for whatever reason than the path to the new config file is what should be supplied with the `--config` option.